See https://github.com/python/cpython/pull/23762

Should be fixed in the next Python 3.9.x release at which point we can drop
the patch.

diff -wpruN '--exclude=*.orig' a~/Modules/_posixsubprocess.c a/Modules/_posixsubprocess.c
--- a~/Modules/_posixsubprocess.c	1970-01-01 00:00:00
+++ a/Modules/_posixsubprocess.c	1970-01-01 00:00:00
@@ -753,7 +753,7 @@ subprocess_fork_exec(PyObject* self, PyO
     if (groups_list != Py_None) {
 #ifdef HAVE_SETGROUPS
         Py_ssize_t i;
-        unsigned long gid;
+        gid_t gid;
 
         if (!PyList_Check(groups_list)) {
             PyErr_SetString(PyExc_TypeError,
